### PR TITLE
fix: fix error in using relativeHeadRowIndex for writing tables only

### LINKED ISSUE #794: [Bug] 使用table的relativeHeadRowIndex()设置写入行偏移量时效果不对
### Search before asking

- [x] I searched in the [issues](https://github.com/apache/fesod/issues) and found nothing similar.

### Fesod version

1.3.0

### JDK version

Microsoft OpenJDK 17.0.17

### Operating system

Windows11 专业版 24H2

### Steps To Reproduce

Image
Image

### Current Behavior

实际上第一个对象的数据头在第6行，数据从第12行开始
Image

### Expected Behavior

预期第一个对象的数据头在第6行，数据从第7行开始
Image

### Anything else?

在添加数据时，会初始化当前table
Image
在初始话当前table的时候会写入设置的head，并且从指定的偏移量开始，我这里是5，所以从第6行开始，这是对的
Image
紧接着在添加数据的时候，获取到了添加head之后的行数，7，这是对的，但是再一次获取了偏移量，导致了这个错误
Image

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by 32154678925**
## Purpose of the pull request
Closed: #794 
When you do not write directly in the `sheet`, but only write data in the `table `and set the `head`, the offset value set with `relativeHeadRowIndex()` will make an error, and the blank line with the specified offset will be empty again between the head and the data. If the `head `is not set, the offset value set with `relativeHeadRowIndex()` will not take effect

## What's changed?
Change the judgment of whether to add offset in the `add` method in the `ExcelWriteAddExecutor` class from `writeSheetHolder` to `currentWriteHolder`

**Before**:
```java
WriteSheetHolder writeSheetHolder = writeContext.writeSheetHolder();
int newRowIndex = writeSheetHolder.getNewRowIndexAndStartDoWrite();
if (writeSheetHolder.isNew()
        && !writeSheetHolder.getExcelWriteHeadProperty().hasHead()) {
    newRowIndex += writeContext.currentWriteHolder().relativeHeadRowIndex();
}
```

**After**:
```java
WriteSheetHolder writeSheetHolder = writeContext.writeSheetHolder();
WriteHolder currentWriteHolder = writeContext.currentWriteHolder();
int newRowIndex = writeSheetHolder.getNewRowIndexAndStartDoWrite();
if (currentWriteHolder.isNew()) {
    AbstractWriteHolder writeHolder = (AbstractWriteHolder) currentWriteHolder;
    if(!writeHolder.getExcelWriteHeadProperty().hasHead()){
        newRowIndex += currentWriteHolder.relativeHeadRowIndex();
    }
}
```

## Checklist

- [x] I have read the [Contributor Guide](https://fesod.apache.org/community/contribution/).
- [ ] I have written the necessary doc or comment.
- [x] I have added the necessary unit tests and all cases have passed.

### CODE REVIEW

**Comment by bengbengbalabalabeng** [comment]
These tests contain no assertions and therefore do not validate the fix. Please enhance them with assertions that check the actual output. (Also ensure that both XLSX and XLS formats are covered)

**Comment by 32154678925** [comment]
> 这些测试不包含断言，因此不验证修复。请用检查实际输出的断言来增强它们。（同时确保涵盖XLSX和XLS格式）

谢谢，我修改了测试代码，并且成功跑通测试，且测试了未修改bug的代码，测试是跑不同的。我的代码经验不多，还有什么需要修改的么？

Thank you. I modified the test code and successfully ran through the test. I also tested the code without modifying the bug. The test runs differently. I don't have much code experience. Is there anything I need to change?

File: fesod-sheet/src/test/java/org/apache/fesod/sheet/writesheet/WriteTableTest.java [hunk 1]
```diff
@@ -0,0 +1,247 @@
+package org.apache.fesod.sheet.writesheet;
```
Comment (bengbengbalabalabeng) [inline fesod-sheet/src/test/java/org/apache/fesod/sheet/writesheet/WriteTableTest.java]: Please supplement the copyright header. [license-header.txt](https://github.com/apache/fesod/blob/main/tools/spotless/license-header.txt)

File: fesod-sheet/src/test/java/org/apache/fesod/sheet/writesheet/WriteTableTest.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            throw new RuntimeException("Failed to resolve @ExcelProperty header from WriteSheetData.string", e);
+        }
+    }
+
+    @ParameterizedTest
+    @EnumSource(
+            value = ExcelTypeEnum.class,
+            names = {"XLS", "XLSX"})
+    public void testWriteTableWithTitle(ExcelTypeEnum excelType) throws Exception {
+        Random random = new Random();
```
Comment (bengbengbalabalabeng) [inline fesod-sheet/src/test/java/org/apache/fesod/sheet/writesheet/WriteTableTest.java]: Random numbers may lead to non-deterministic, it is recommended to use @MethodSource to provide N sets of representative boundary values for testing.

**Comment by bengbengbalabalabeng** [comment]
Suggest covering multi-sheet table writing scenarios.

**Comment by 32154678925** [comment]
> Suggest covering multi-sheet table writing scenarios.

我添加了版权声明，移除了随机数，添加了多组偏移量，它们现在能够正常工作，还有什么需要修改的么？

I added a copyright notice, removed random numbers, and added multiple sets of offsets. They now work properly. Is there anything else I need to modify?

File: fesod-sheet/src/test/java/org/apache/fesod/sheet/writesheet/WriteTableTest.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.fesod.sheet.writesheet;
+
+import java.io.File;
+import java.util.*;
```
Comment (bengbengbalabalabeng) [inline fesod-sheet/src/test/java/org/apache/fesod/sheet/writesheet/WriteTableTest.java]: Avoid using `import *` (34 lines have the same issue)

Comment (32154678925) [reply fesod-sheet/src/test/java/org/apache/fesod/sheet/writesheet/WriteTableTest.java]: > Avoid using `import *` (34 lines have the same issue)

我已经修改了引入规则，还有其他需要注意的么？或者本项目有什么规则之类的说明？或许我可以去看看。

I have modified the import rules. Is there anything else I should pay attention to? Or what are the rules of this project? Maybe I can go and have a look.

Comment (bengbengbalabalabeng) [reply fesod-sheet/src/test/java/org/apache/fesod/sheet/writesheet/WriteTableTest.java]: The unit tests for the `fesod-sheet` module have recently undergone systematic refactoring, and the original `TestFileUtil` has been completely removed. It is recommended to refer to the following latest test case examples:

`org.apache.fesod.sheet.sheet.WriteSheetTest`
`org.apache.fesod.sheet.sheet.HiddenSheetsTest`
......

Comment (32154678925) [reply fesod-sheet/src/test/java/org/apache/fesod/sheet/writesheet/WriteTableTest.java]: I remove import TestFileUtil

**Review by bengbengbalabalabeng — APPROVED** [review]
LGTM.

Merge in 24h.
